Sources/ArcGISToolkit/Components/LocationButton/LocationButton.swift

```diff
@@ -0,0 +1,342 @@
+    /// The next auto pan mode to be used when cycling through auto pan modes.
+    func nextAutoPanMode(current: LocationDisplay.AutoPanMode, initial: LocationDisplay.AutoPanMode) -> LocationDisplay.AutoPanMode {
+        guard let index = autoPanOptions.firstIndex(of: current) else { return initial }
+        let nextIndex = index.advanced(by: 1) == autoPanOptions.endIndex ? autoPanOptions.startIndex : index.advanced(by: 1)
+        return autoPanOptions[nextIndex]
+    }
+    
+    /// Hides the location display.
+    func hideLocationDisplay() async {
+        await locationDisplay.dataSource.stop()
+    }
+}
+
+extension LocationButton {
+    /// The type of actions that can take place when the button is pressed.
+    enum Action {
+        /// Start the location display.
+        case start
+        /// Stop the location display.
+        case stop
+        /// Set the next auto pan mode for cycling through.
+        case autoPanCycle
+        
+        /// The action that should occur for the specified state.
+        init?(
+            status: LocationDataSource.Status,
+            autoPanOptions: [LocationDisplay.AutoPanMode]
+        ) {
+            // Decide the button behavior based on the status.
+            switch status {
+            case .stopped, .failedToStart:
+                self = .start
+            case .started:
+                if autoPanOptions.count < 2 {
+                    // If there were no non-off options specified then set it to off
+                    // if the status is started.
+                    self = .stop
+                } else {
+                    self = .autoPanCycle
+                }
+            case .starting, .stopping:
+                return nil
+            @unknown default:
+                fatalError()
+            }
+        }
+    }
+}
+
+private extension LocationDisplay.AutoPanMode {
+    /// The label that should appear in the picker.
+    var pickerText: String {
+        switch self {
+        case .off:
+            "Auto Pan Off"
+        case .recenter:
+            "Recenter"
+        case .compassNavigation:
+            "Compass"
+        case .navigation:
+            "Navigation"
+        @unknown default:
+            fatalError()
+        }
+    }
+    
+    /// The image associated with the auto pan mode.
+    var imageSystemName: String {
+        switch self {
+        case .compassNavigation:
+            "field.of.view.wide.fill"
+        case .navigation:
+            "location.north.fill"
+        case .recenter:
+            "location.fill.viewfinder"
+        case .off:
+            "location"
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+private extension Sequence where Iterator.Element: Hashable {
+    /// Returns the unique values of the sequence with the order preserved.
+    func unique() -> [Iterator.Element] {
+        var seen: Set<Iterator.Element> = []
+        return filter { seen.insert($0).inserted }
+    }
+}
```

Comment:
Swift Algorithms provides a `uniqued()` [method](https://swiftpackageindex.com/apple/swift-algorithms/1.2.1/documentation/algorithms/swift/sequence/uniqued()). If you haven't done so yet, consider adding Swift Algorithms as a dependency and using its uniquing algorithm.

Reply:
will do that